src/connection.ts

```diff
@@ -39,6 +39,7 @@ export const SFDX_HTTP_HEADERS = {
 };
 
 export const DNS_ERROR_NAME = 'Domain Not Found';
+const DNS_TIMEOUT = Math.min(3, new Env().getNumber('SFDX_DNS_TIMEOUT') || 3);
```

Comment:
@mshanemc should be `Math.max`. Also getNumber supports supplying a default with second parameter.